src/ie9-ajax.js

```diff
@@ -58,8 +58,17 @@ function ie9Ajax(method, url, opts) {
     xdr.onerror = handleError;
     xdr.ontimeout = handleError;
 
-    xdr.open(method, authToUrl(url, opts));
-    xdr.send(opts.data);
+    try {
+      xdr.open(method, authToUrl(url, opts));
+      xdr.send(opts.data);
+    } catch (error) {
+      // using trim because the error message has a bunch of white space for some reason
+      if (error.message.trim() === 'Access is denied.') {
+        reject(new Error('Please ensure Shopify JavaScript Buy SDK is run over https'));
```

Comment:
With the logger implemented the user will have the following output:
`Unhandled promise rejectionError: Please ensure Shopify JavaScript Buy SDK is run over https`

Implemented here:
https://github.com/Shopify/js-buy-sdk/blob/feature/ie9-error-messaging/src/ie9-ajax.js#L69